src/payment/external-id-request-sender.ts

```diff
@@ -0,0 +1,25 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { ContentType, INTERNAL_USE_ONLY } from '../common/http-request';
+
+export default class ExternalIdRequestSender {
+    constructor(
+        private _requestSender: RequestSender
+    ) {}
+
+    async saveExternalId(provider: string, externalId: string): Promise<void> {
+        const url = `/api/storefront/payment/${provider}/save-external-id`;
+        const options = {
+            headers: {
+                Accept: ContentType.JsonV1,
+                'X-API-INTERNAL': INTERNAL_USE_ONLY,
+            },
+            body: {
+                externalId,
+                provider,
+            },
+        };
+
+        await this._requestSender.post<''>(url, options);
```

Comment:
should this be `.post<void>`? Otherwise, it's implying an empty string is returned, which is not very valuable anyway.